operator/internal/manifests/config_otlp.go

```diff
@@ -0,0 +1,273 @@
+		slices.Sort(attrs[0].Names)
+		attrs[0].Names = slices.Compact(attrs[0].Names)
+	}
+
+	if len(attrs) == 1 {
+		// If there's only one item, then skip the complex sorting
+		return attrs
+	}
+
+	slices.SortFunc(attrs, func(a, b config.OTLPAttribute) int {
+		action := strings.Compare(string(a.Action), string(b.Action))
+		if action != 0 {
+			return action
+		}
+
+		if a.Regex != "" && b.Regex != "" {
+			return strings.Compare(a.Regex, b.Regex)
+		}
+
+		if a.Regex != "" && b.Regex == "" {
+			return 1
+		}
+
+		if a.Regex == "" && b.Regex != "" {
+			return -1
+		}
+
+		return 0
+	})
+
+	for i := 0; i < len(attrs)-1; i++ {
+		a := attrs[i]
+		if a.Regex != "" {
+			continue
+		}
+
+		slices.Sort(a.Names)
+		attrs[i] = a
+
+		next := attrs[i+1]
+		if next.Regex != "" {
+			continue
+		}
+
+		if a.Action != next.Action {
+			continue
+		}
+
+		// Combine attribute definitions if they have the same action and just contain names
+		a.Names = append(a.Names, next.Names...)
+		slices.Sort(a.Names)
+		a.Names = slices.Compact(a.Names)
+
+		// Remove the "next" attribute definition
+		attrs[i] = a
+		attrs = append(attrs[:i+1], attrs[i+2:]...)
+		i--
+	}
+
+	return attrs
+}
+
+func otlpAttributeConfig(ls *lokiv1.LokiStackSpec) config.OTLPAttributeConfig {
+	result := defaultOTLPAttributeConfig(ls.Tenants)
+
+	if ls.Limits != nil {
+		if ls.Limits.Global != nil && ls.Limits.Global.OTLP != nil {
+			result.RemoveDefaultLabels = true
+			globalOTLP := ls.Limits.Global.OTLP
+
+			if streamLabels := globalOTLP.StreamLabels; streamLabels != nil {
+				if result.Global == nil {
+					result.Global = &config.OTLPTenantAttributeConfig{}
+				}
+
+				if resAttr := streamLabels.ResourceAttributes; len(resAttr) > 0 {
+					result.Global.ResourceAttributes = append(result.Global.ResourceAttributes,
+						convertAttributeReferences(resAttr, config.OTLPAttributeActionStreamLabel)...)
+				}
+			}
+
+			if structuredMetadata := globalOTLP.StructuredMetadata; structuredMetadata != nil {
+				if result.Global == nil {
+					result.Global = &config.OTLPTenantAttributeConfig{}
+				}
+
+				if resAttr := structuredMetadata.ResourceAttributes; len(resAttr) > 0 {
+					result.Global.ResourceAttributes = append(result.Global.ResourceAttributes,
+						convertAttributeReferences(resAttr, config.OTLPAttributeActionMetadata)...)
+				}
+
+				if scopeAttr := structuredMetadata.ScopeAttributes; len(scopeAttr) > 0 {
+					result.Global.ScopeAttributes = append(result.Global.ScopeAttributes,
+						convertAttributeReferences(scopeAttr, config.OTLPAttributeActionMetadata)...)
+				}
+
+				if logAttr := structuredMetadata.LogAttributes; len(logAttr) > 0 {
+					result.Global.LogAttributes = append(result.Global.LogAttributes,
+						convertAttributeReferences(logAttr, config.OTLPAttributeActionMetadata)...)
+				}
+			}
+		}
+
+		for tenant, tenantLimits := range ls.Limits.Tenants {
+			if tenantLimits.OTLP != nil {
+				result.RemoveDefaultLabels = true
+
```

Comment:
AFAIU removing the default labels from the general distibutor otlp config means that we can construct a loki config set (config+runtime-config) without indexing any labels for **non-openshift** tenant modes. I tried this with the following on top of the `hack/lokistack_gateway_dev.yaml`:

```yaml
  limits:
    tenants:
      test-oidc:
        otlp:
          structuredMetadata:
            resourceAttributes:
            - name: stream.test.label
```

and results in :
```yaml
distributor:
  otlp_config:
    default_resource_attributes_as_index_labels: []
```

and:

```yaml
overrides:
  test-oidc:
    otlp_config:
      resource_attributes:
        attributes_config:
        - action: structured_metadata
          attributes:
          - stream.test.label
```

Is this a case for validation (min one stream label given) or do we need to make `result.RemoveDefaultLabels = true` depending on tenant modes?